pkg/api/bzz.go

```diff
@@ -260,6 +256,16 @@ func (s *server) bzzDownloadHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	s.serveReference(address, pathVar, w, r)
+}
+
+func (s *server) serveReference(address swarm.Address, pathVar string, w http.ResponseWriter, r *http.Request) {
```

Comment:
@mrekucci So this method is now a helper which can be called from different callers. Ideally we would want to have the logs tagged with the parent request, so basically distinguish if we are serving on `/bzz` or on subdomain. Would this be something handled better with you new logging handler?

Right now we can pass a logger instance to this method which can be prefixed with some string based on the caller. But then we would need to have some basic proxy implementation of the logger to add the prefix.